### DISCUSSION TITLE
how to connect to ftp server to upload pb_hooks files

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pockethost/pockethost#352] [Q&A] [answered]
command using : wget --ftp-user=userName --ftp-password=pass ftp://ftp.sfo-1.pockethost.io   

error: PASV ... couldn't connect to 0.0.0.0 port 10019: Connection refused

**Comment by nikhil-varma-mudunuri** [comment]
able to connect using lftp 
but getting below error when using ls
ls: Fatal error: Certificate verification: certificate has expired (F2:B1:FF:0F:54:94:DB:BB:91:F7:A3:A9:8E:C3:C8:DB:DA:A1:12:54)

**Reply by benallfree** [reply]
I will fix this. In the mean time, does `lftp` have a way to bypass the expired cert error?

**Reply by benallfree** [reply]
Should be fixed 👍

**Comment by nikhil-varma-mudunuri** [comment]
able to connect but , unable to execute ls command 

421 closing connection error

it will be really easy if you provide ssh access to the user, for uploading files

**Reply by benallfree** [reply]
Hmm, is that happening consistently?

SSH access could be a good idea - we are accepting PR's :)

**Comment by nikhil-varma-mudunuri** [comment]
yes constantly getting error, ssh will be a good way to upload the files as it was widely used and easy to understand (in my option)

**Reply by benallfree** [reply]
Do you get the same error using FileZilla?

**Reply by nikhil-varma-mudunuri** [reply]
ha yes
getting below error:

Error:         	Failed to retrieve directory listing
Response: 	421 Closing connection
Error:         	GnuTLS error -110 in gnutls_record_recv: The TLS connection was non-properly terminated.
Status:      	Server did not properly shut down TLS connection
Error:         	Could not read from socket: ECONNABORTED - Connection aborted
Error:         	Disconnected from server

any work around for transfering files, need quickly plzzzz

**Reply by benallfree** [reply]
Sorry you're still having trouble :( 

Please check these settings:

<img width="523" alt="image" src="https://github.com/pockethost/pockethost/assets/1068356/ed497e79-a0c2-4aa9-90d1-c674c996169a">

Then, paste the entire FileZilla session, like this:

```
Status:      	Resolving address of ftp.sfo-1.pockethost.io
Status:      	Connecting to 147.182.196.168:21...
Status:      	Connection established, waiting for welcome message...
Status:      	Initializing TLS...
Status:      	TLS connection established.
Status:      	Logged in
Status:      	Retrieving directory listing...
Status:      	Server sent passive reply with unroutable address. Using server address instead.
Status:      	Directory listing of "/" successful
```

**Reply by nikhil-varma-mudunuri** [reply]
sorry , for late reply
getting different error when i changed protocol to sftp from ftp

Error:         	FATAL ERROR: No supported authentication methods available (server sent: publickey)
Error:         	Could not connect to server

**Reply by benallfree** [reply, marked as answer]
@nikhil-varma-mudunuri Please come chat about this with us on https://discord.com/channels/1128192380500193370/1128192380500193373, it's going to take too long to do it this way.

If you can't chat on Discord, please make sure you are using your pockethost login and password, not SSH keys.

**Comment by nikhil-varma-mudunuri** [comment]
when can i except ssh access to the server , if it is possible

**Reply by benallfree** [reply]
PocketHost is an open source project and you are invited to submit pull requests for features such as SSH. 

SSH is on the roadmap but there is currently no timeline because it has some hidden complexities. It’s a popular idea though.
